ctwrap/simulation.py

```diff
@@ -634,6 +634,15 @@ def run_parallel(self,
             if verbosity > 1:
                 print(indent2 + msg)
 
+        if verbosity > 1:
+            print("Appending metadata")
+
+        try:
```

Comment:
Why are you using a `try` block here - I believe the `except` portion just re-raises the exception as if you didn't use the `try`-`except` structure in the first place?

Writing meta data outside the worker is a good idea; however, isn't there already a `Simulation` object available - i.e. `sim` (which is passed as an argument)?